bolt-design-system/bolt
# Devops Updates

## Jira
http://vjira2:8080/browse/BDS-1356

## Summary
Assortment of minor DevOps-related Bolt updates -- mostly relating to updating NPM dependencies globally, fixing a few bugs with the API rendering service, smoothing over some issues with yarn installs, and adding a new loading screen when booting up a new dev server.

## Details
- Globally upgraded NPM package dependencies for the monorepo, docs site, build tools, shared configs, and Bolt core.
- Fixed an unexpected regression from #1178 causing components being rendered via the Component Explorer to be partially missing HTML in the response
- Updated the @bolt/build-tools to re-introduce a new and improved loading screen -- now working with webpack-dev-server!
![bolt-loading-screen](https://user-images.githubusercontent.com/1617209/58340867-dcca3680-7e1a-11e9-987c-7025f0af358d.gif)
- Monorepo Node.js version upgraded to the latest stable version (v10.15.3)
- Significantly fewer warnings when installing and running the Bolt monorepo locally
- Fixed a minor CSS layout issue causing unnecessary scrolling on the docs site homepage
- Upgrade the version / package name of the lit-element dependency getting used / tested out. Addresses #1162
- Removes all but one instance where we specify a twig/twig dependency in any of our composer.json files (now only specified in our Twig renderer package). Not only should this help make it easier to manage which version gets used in the monorepo, but this should also help decouple which version of Twig ultimately gets used downstream, hopefully helping to address issues flagged like #1175
- Updated the docs site-specific `.boltrc` config to automatically open a new tab when booting up a new dev server instance. 
> As a future update this ideally could be customized per developer's personal preference via a local config file that gets merged with these repo-specific defaults -- one that isn't version controlled  

**Side Note(s):**
- I _think_ these updates should also help to address the annoying #1176 bug involving the `error An unexpected error occurred: "ENOENT: no such file or directory, lstat '/Users/USERNAME/sites/bolt-master/packages/build-tools/node_modules/find-up'".` error sporadically getting thrown!
- I'm still not sure if the Sharp SVG-related error will be addressed with these updates BUT I haven't seen that error pop up so far..

## How to test
1. Confirm Jest and Nightwatch tests passing + deploying as expected 
2. Locally everything runs as expected 
3. Gut check with @christophersmith262 or @remydenton that removing the need to specify the Twig version in everything but the Twig renderer (which Drupal doesn’t use) is a better approach to reduce the risk of having different versions of Twig clash
